ansible/roles/create-ecr-build-role/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+- name: check service_id max length
+  fail:
+    msg: "service_id is too long"
+  when: (service_id | length) > 40
+
+- name: check service_id for spaces
+  fail:
+    msg: "service_id contains spaces"
+  when: "' ' in service_id"
+
+- name: check if role exists
+  command: "{{ aws_cmd }} iam get-role --role-name {{ build_role_name }} --query Role"
+  register: get_role
+  changed_when: no
+  failed_when: get_role.rc !=0 and 'NoSuchEntity' not in get_role.stderr
+
+- name: create role
+  block:
+    - name: create temporary file
+      tempfile:
+        state: file
+        suffix: temp
+      register: assume_role_policy_json
+
+    - name: write policy json
+      copy:
+        content: "{{ assume_role_policy | to_json }}"
+        dest: "{{ assume_role_policy_json.path }}"
+
+    - name: create iam role
+      command: >-
+        {{ aws_cmd }} iam create-role --role-name {{ build_role_name }}
+        --assume-role-policy-document file://{{ assume_role_policy_json.path }}
+        --query Role
+      register: create_role
+      when: get_role.rc == 255
+
+  always:
+    - name: remove temp file
+      file:
+        path: "{{ assume_role_policy_json.path }}"
+        state: absent
+      when: assume_role_policy_json.path is defined
+
+- name: parse role json
+  set_fact:
+    iam_role: "{{ (create_role.stdout if get_role.rc == 255 else get_role.stdout)  | from_json }}"
+
+- name: get build role policy
```

Comment:
Is there an AWS ansible plugin we could be using instead of calling the CLI directly? Might save some checking rcs.

Reply:
yes / no .. it tends to be out of date of the latest functionality ( last time I looked it didn't have secretsmanager ..  ) def a potential tech debt ticket, to look into it